README.md

```diff
@@ -1,32 +1,120 @@
+
+### Lasso Shared Controller
+The sharedController type embeds the Controller interface to expose it's controller's functionality. The sharedController also exposes the RegisterHandler function along with the Client function. RegisterHandler adds the passed handler to the controller's handler list and client returns a client for the sharedController corresponding resource type; sharedController has a GVK field (Group Version Kind) which indicates what type it is for.
+
+The sharedController can create its underlying controller. It will only do so once a handler is registered to the sharedController.
+
+### Lasso Client
+Wraps Kubernetes client. Contains GVK fields for managing the client in a parent struct such as the sharedClientyFactory.
+
+### Lasso Cache
+The lasso Cache is a light wrapper around around a [SharedIndexInformer from client-go](https://pkg.go.dev/k8s.io/client-go/tools/cache#SharedIndexInformer).
+
+### Lasso Shared Cache Factory
+The shared cache factory stores the Caches and allows them to be managed via GVK structs. Once the Caches have been created once they can be reused.
+
+### Lasso Shared Client Factory
+The Shared Client Factory exists so that after a client for a GVK has been created once it can be reused.
+
+### Lasso Shared Controller Factory
+The SharedControllerFactory ties everything above together. It allows for a single entry point to access clients, caches, and handlers- all of which it manages for you. It is possible to mainly interact with the shared controller factory. It will create all the underlying controllers, clients, and caches you need. It will ensure there are not duplicates of these objects. It will prevent race conditions.
+
+Once you use ForObject, ForKind, ForResource, or ForResourceKind, the SharedControllerFactory will do the following:
+![SharedControllerFactory Diagram](docs/assets/SharedControllerFactory.png)
+
+### Deferred Controller Function
+
+The deferredController is a function that once executed will created and return a controller. It will create the controller using a cache from the SharedCacheFactory. The deferredController function's purpose is to hold off on populating the cache until absolutely necessary. This ensures only Kubernetes resources that have registered handlers, been enqueued, or have had their controller's cache requested, are being stored in memory.
```

Comment:
```suggestion
The deferredController is a function that, once executed, will created and return a controller. It will create the controller using a cache from the SharedCacheFactory. The deferredController function's purpose is to hold off on populating the cache until absolutely necessary. This ensures only Kubernetes resources that have registered handlers, been enqueued, or have had their controller's cache requested, are being stored in memory.
```